server/sonar-db-dao/src/main/resources/org/sonar/db/permission/AuthorizationMapper.xml

```diff
@@ -219,17 +219,72 @@
 
   <sql id="sqlSelectPublicProjectsIfRole">
     select
-      p.id
+    p.id
     from
-      projects p
+    projects p
     where
-      <foreach collection="componentIds" open="(" close=")" item="element" index="index" separator=" or ">
-        p.id=#{element,jdbcType=BIGINT}
-      </foreach>
-      and p.private = ${_false}
-      and #{role,jdbcType=VARCHAR} in ('user','codeviewer')
+    <foreach collection="componentIds" open="(" close=")" item="element" index="index" separator=" or ">
+      p.id=#{element,jdbcType=BIGINT}
```

Comment:
Wouldn't `IN` be better than `=`? (Although that change should probably not be part of this commit)